js/bitfinex2.js

```diff
@@ -358,23 +379,68 @@ module.exports = class bitfinex2 extends bitfinex {
         return this.parseTicker (ticker, market);
     }
 
-    parseTrade (trade, market) {
-        let [ id, timestamp, amount, price ] = trade;
-        let side = (amount < 0) ? 'sell' : 'buy';
-        if (amount < 0) {
-            amount = -amount;
+    parseTrade (trade, market = undefined) {
+        if (trade.length > 5) {
+            // 'my' trades
+            let id = trade[0];
+            let pair = trade[1];
+            market = this.markets_by_id[pair];
+            // todo: need to figure out how to map old BCH pairs
+            let symbol = market ? market['symbol'] : pair;
+            let mtsCreate = trade[2];
+            let orderId = trade[3];
+            let execAmount = trade[4];
+            let execPrice = trade[5];
+            let orderType = trade[6];
+            // let orderPrice = trade[7];
+            let maker = trade[8];
+            let fee = -trade[9];
+            let feeCurrency = trade[10];
+            let type = this.options.orderTypes[orderType];
```

Comment:
This line won't work in Python, because `this.options` is a dict, and you can't dereference a key inside a dict in Python like you're used to to in JS, this violates a rule from the following list: https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#derived-exchange-classes

> all associative keys must be single-quoted strings everywhere, `array['good']`, `array.bad`